new-log-viewer/src/services/decoders/JsonlDecoder/utils.ts

```diff
@@ -0,0 +1,85 @@
+import dayjs, {Dayjs} from "dayjs";
+
+import {
+    JsonObject,
+    JsonValue,
+} from "../../../typings/js";
+import {
+    INVALID_TIMESTAMP_VALUE,
+    LOG_LEVEL,
+} from "../../../typings/logs";
+
+
+/**
+ * Determines whether the given value is a `JsonObject` and applies a TypeScript narrowing
+ * conversion if so.
+ *
+ * @param value
+ * @return A TypeScript type predicate indicating whether `value` is a `JsonObject`.
+ */
+const isJsonObject = (value: JsonValue): value is JsonObject => {
+    return "object" === typeof value && null !== value && !Array.isArray(value);
+};
+
+/**
+ * Converts a field into a log level if possible.
+ *
+ * @param field
+ * @return The log level or `LOG_LEVEL.NONE` if the field couldn't be converted.
+ */
+const convertToLogLevelValue = (field: JsonValue | undefined): LOG_LEVEL => {
+    let logLevelValue = LOG_LEVEL.NONE;
+
+    // If the field is an object , e.g. `field = { "name": "INFO", "value": 20 }`.
+    // Then the user should specify a nested key.
+    if ("undefined" === typeof field || "object" === typeof field) {
```

Comment:
Shall we use `isJsonObject`? Otherwise it's possible that `field` is an array or null.